Authorization/Authorization/Presentation/Login/SignInView.swift

```diff
@@ -16,137 +16,170 @@ public struct SignInView: View {
     @ObservedObject
     private var viewModel: SignInViewModel
     
-    public init(viewModel: SignInViewModel) {
+    private var navigationController: UINavigationController
+    
+    public init(viewModel: SignInViewModel, navigationController: UINavigationController) {
         self.viewModel = viewModel
+        self.navigationController = navigationController
+    }
+
+    public func webLogin() async {
+        await viewModel.login(viewController: self.navigationController)
     }
     
     public var body: some View {
-        ZStack(alignment: .top) {
-            VStack {
-                CoreAssets.authBackground.swiftUIImage
-                    .resizable()
-                    .edgesIgnoringSafeArea(.top)
-            }.frame(maxWidth: .infinity, maxHeight: 200)
-            
-            VStack(alignment: .center) {
-                CoreAssets.appLogo.swiftUIImage
-                    .resizable()
-                    .frame(maxWidth: 189, maxHeight: 54)
-                    .padding(.vertical, 40)
+        if viewModel.forceWebLogin {
+            // On first load, we should bring the user right to the web login
+            // interface rather than showing this view.
+            //
+            // If that login fails or the user escapes back, they'll be brought
+            // back to the view where any error message will be shown.
+            Task {
+                await webLogin()
+            }
+        }
```

Comment:
In my mind, a more SwiftUI way will put this code to the View `.onAppear {}`
What do you think about it?

Reply:
@volodymyr-chekyrta I'm quite new to Swift, SwiftUI, and its idioms. This is my first Swift code, in fact :)

So, I couldn't tell you for sure. But I've looked up the docs, and I think you're probably right! I'll add this.